src/Compilers/CSharp/Portable/Emitter/Model/PEModuleBuilder.cs

```diff
@@ -790,6 +790,19 @@ protected virtual Cci.IModuleReference TranslateModule(ModuleSymbol module, Diag
             {
                 Debug.Assert(!needDeclaration);
                 namedTypeSymbol = namedTypeSymbol.TupleUnderlyingType;
+
+                // check that underlying type of a ValueTuple is indeed a value type (or error)
+                // this should never happen, in theory,
+                // but if it does happen we should make it a failure.
+                var declaredBase = namedTypeSymbol.BaseTypeNoUseSiteDiagnostics;
+                if (declaredBase.SpecialType != SpecialType.System_ValueType && !declaredBase.IsErrorType())
```

Comment:
namedTypeSymbol is not necessarily a tuple and many types would default to be classes or whatever.
We do not want Enums, which are value types, so checking for a base ValueType is actually simpler and more precise.

---
In reply to: [81642661](https://github.com/dotnet/roslyn/pull/14161#discussion_r81642661) [](ancestors = 81642661)